static/app/components/events/autofix/autofixInsightCards.tsx

```diff
@@ -117,6 +119,31 @@ function AutofixInsightCard({
 
   const insightCardAboveIndex = index - 1 >= 0 ? index - 1 : null;
 
+  // Typing animation for new insight
+  useEffect(() => {
+    if (!isNewInsight) {
+      setDisplayedInsightTitle(insight.insight);
+      titleIndexRef.current = insight.insight.length;
+      return () => {};
+    }
+
+    setDisplayedInsightTitle('');
+    titleIndexRef.current = 0;
+
+    const intervalId = window.setInterval(() => {
+      if (titleIndexRef.current < insight.insight.length) {
+        setDisplayedInsightTitle(insight.insight.slice(0, titleIndexRef.current + 1));
+        titleIndexRef.current++;
+      } else {
+        window.clearInterval(intervalId);
+      }
+    }, 15);
+
+    return () => {
+      window.clearInterval(intervalId);
+    };
+  }, [insight.insight, isNewInsight]);
```

Comment:
thinking out loud here, but we use a typing animation like this in multiple places, might be worth making a component with this logic that can be reusable